### PR TITLE
[docs] Automatic tablet splitting update

### PR DESCRIPTION
**Pull request by stevebang**
Adds:

- content and example for automatic tablet splitting to architecture page
- adds missing `yb-admin split_tablet` command
- Edited and fixed lots of typos on tablet splitting architecture page.
- Added cross-references from yb-admin and yb-master to relevant docs.

@ttyusupov is adding some notes and usage information. Search for `???` to find content needed and questions.

### CODE REVIEW

File: docs/content/latest/architecture/docdb-sharding/tablet-splitting.md [hunk 1]
```diff
@@ -195,21 +192,112 @@ Tablet UUID                       Range                                 Leader
...
+
+For details on the architecture design document, see [Automatic Re-sharding of Data with Tablet Splitting]](https://github.com/yugabyte/yugabyte-db/blob/master/architecture/design/docdb-automatic-tablet-splitting.md). While the broader feature is [work-in-progress](https://github.com/yugabyte/yugabyte-db/issues/1004), tablets can be automatically split for a few scenarios starting in the v2.2 release.
+
+??? Which scenarios are the "few scenarios" listed in the previous sentence?
+
+### Enable automatic tablet splitting
+
+To enable automatic tablet splitting, use the `yb-master` [`--tablet_split_size_threshold_bytes`](../../reference/configuration/yb-master/#tablet-split-size-threshold-bytes) flag to specify the size when tablets should split.
+
+??? Need guidance and tips on what values to use.
```
Comment (bmatican) [inline docs/content/latest/architecture/docdb-sharding/tablet-splitting.md:220]: cc @ttyusupov

File: docs/content/latest/architecture/docdb-sharding/tablet-splitting.md [hunk 2]
```diff
@@ -195,21 +192,112 @@ Tablet UUID                       Range                                 Leader
...
+(curl -s http://127.0.0.1:9000/metrics && curl -s http://127.0.0.2:9000/metrics && curl -s http://127.0.0.3:9000/metrics) | jq -s '[flatten | .[] | select(.type == "tablet") as $tablet | .metrics[] | select(.name == "rocksdb_number_db_seek" and .value > 0) | { tablet_id: $tablet.id, rocksdb_number_db_seek: .value }] | sort_by (.tablet_id)' >after-run.json
+```
+
+9. Get the list of tablets accessed during the run phase of the workload.
+
+```sh
+diff -C1 after-load.json after-run.json | grep tablet_id | sort | uniq
+```
+
+10. The list of tablets accessed can be compared with `http://127.0.0.1:9000/tablets` to make sure no pre-split tablets have been accessed.
```
Comment (bmatican) [inline docs/content/latest/architecture/docdb-sharding/tablet-splitting.md:302]: we also need a sub section of extra notes, eg: tablets will not split again until a full compaction, etc cc @ttyusupov

File: docs/content/latest/architecture/docdb-sharding/tablet-splitting.md [hunk 3]
```diff
@@ -12,67 +12,70 @@ isTocNested: true
 showAsideToc: true
 ---
 
-Tablet splitting enables changing the number of tablets at runtime. This enables *re-shrading* the existing data in the cluster.
+Tablet splitting enables changing the number of tablets at runtime. This enables *resharding* the existing data in the cluster.
```
[the remark this answers is not in the document]
Comment (stevebang) [reply docs/content/latest/architecture/docdb-sharding/tablet-splitting.md:15]: Fixed that too, after double-checking dictionaries.

File: docs/content/latest/admin/yb-admin.md [hunk 4]
```diff
@@ -240,6 +240,28 @@ yb-admin -master_addresses <master-addresses> list_tablets_for_tablet_server <ts
 - *master-addresses*: Comma-separated list of YB-Master hosts and ports. Default value is `localhost:7100`.
 - *ts_uuid*: The UUID of the tablet server (YB-TServer).
 
+#### split_tablet
+
+Splits the specified hash-partitioned tablet and and computes the split point as the middle of tablet's partition range.
```
Comment (bmatican) [inline docs/content/latest/admin/yb-admin.md:245]: nit: `and and computes`

File: docs/content/latest/admin/yb-admin.md [hunk 5]
```diff
@@ -240,6 +240,28 @@ yb-admin -master_addresses <master-addresses> list_tablets_for_tablet_server <ts
...
+
+{{< note title="Note" >}}
+
+The `yb-admin split_tablet` command is not supported for use with range-partitioned tablets. To follow plans on this, see [GitHub #5166](https://github.com/yugabyte/yugabyte-db/issues/5166)
```
Comment (bmatican) [inline docs/content/latest/admin/yb-admin.md:249]: nit: maybe `not yet`?

File: docs/content/latest/reference/configuration/yb-master.md [hunk 6]
```diff
@@ -381,7 +387,12 @@ yb-admin --master_addresses <master-addresses> --tablet_split_size_threshold_byt
 ```
 
 - *master-addresses*: Comma-separated list of YB-Master hosts and ports. Default value is `localhost:7100`.
-- *bytes*: The threshold size, in bytes, before each tablet should be split. Default value is `0`, Default value of `0` disables dynamic tablet splitting.
+- *bytes*: The threshold size, in bytes, before each tablet should be split. Default value of `0` disables automatic tablet splitting.
```
Comment (bmatican) [inline docs/content/latest/reference/configuration/yb-master.md:390]: `after which tablets should be split`, not before

File: docs/content/latest/admin/yb-admin.md (hunk 4 above)
Comment (stevebang) [reply docs/content/latest/admin/yb-admin.md:245]: fixed

File: docs/content/latest/architecture/docdb-sharding/tablet-splitting.md [hunk 7]
```diff
@@ -12,67 +12,74 @@ isTocNested: true
 showAsideToc: true
 ---
 
-Tablet splitting enables changing the number of tablets at runtime. This enables *re-shrading* the existing data in the cluster.
-
 ## Overview
 
-There are a number of scenarios where this is useful.
+Tablet splitting enables *resharding* existing data in the cluster by changing the number of tablets at runtime. Here are some of the scenarios where this is useful.
```
Comment (bmatican) [inline docs/content/latest/architecture/docdb-sharding/tablet-splitting.md:17]: technically, presplitting doesn't fit in the category of changing things at runtime, as it's done in advance

File: docs/content/latest/admin/yb-admin.md (hunk 5 above)
Comment (stevebang) [reply docs/content/latest/admin/yb-admin.md:249]: Added yet

File: docs/content/latest/architecture/docdb-sharding/tablet-splitting.md [hunk 8]
```diff
@@ -12,67 +12,74 @@ isTocNested: true
...
+* **[Manual tablet splitting](#manual-tablet-splitting):** The tablets in a running cluster can be split manually at runtime by the user.
 
-* **Pre-splitting** All tables created in DocDB can be split into the desired number of tablets at creation time.
+* **[Automatic tablet splitting](#automatic-tablet-splitting-beta):** The tablets in a running cluster are automatically split according to some policy by the database.
 
-* **Manual splitting** The tablets in a running cluster can be split manually at runtime by the user.
+The following sections give details on how to split tablets using these three approaches.
 
-* **Dynamic splitting** The tablets in a running cluster are automatically split according to some policy by the database.
+## Presplitting tablets [BETA]
```
Comment (bmatican) [inline docs/content/latest/architecture/docdb-sharding/tablet-splitting.md:53]: this is not BETA

File: docs/content/latest/reference/configuration/yb-master.md (hunk 6 above)
Comment (stevebang) [reply docs/content/latest/reference/configuration/yb-master.md:390]: good catch -- didn't deal as planned with whether it is after or at that size

File: docs/content/latest/architecture/docdb-sharding/tablet-splitting.md [hunk 9]
```diff
@@ -195,21 +206,94 @@ Tablet UUID                       Range                                 Leader
 a89ecb84ad1b488b893b6e7762a6ca2a  key_start: "\177\377" key_end: ""     127.0.0.3:9100
 ```
 
-{{< note title="Note" >}}
+{{< note title="Important" >}}
 
-There are a few important things to note here. 
-* The original tablet `9991368c4b85456988303cd65a3c6503` is no longer present in the system. It has been replaced with two new tablets. 
-* The tablet leaders are now spread across two nodes, in order to evenly balance the tablets for a table across the nodes in the cluster.
+* The original tablet `9991368c4b85456988303cd65a3c6503` no longer exists and has been replaced with two new tablets.
```
Comment (bmatican) [inline docs/content/latest/architecture/docdb-sharding/tablet-splitting.md:211]: Not sure how to mention this, but at least for now, that tablet still exists, until we enhance the code, to get rid of it. It's just that clients won't use it for reading/writing data.

File: docs/content/latest/architecture/docdb-sharding/tablet-splitting.md [hunk 10]
```diff
@@ -195,21 +206,94 @@ Tablet UUID                       Range                                 Leader
...
+Automatic tablet splitting is currently in [BETA](../../../faq/general/#what-is-the-definition-of-the-beta-feature-tag).
 
 {{< /note >}}
 
-Design for automatic splitting can be reviewed on [GitHub](https://github.com/yugabyte/yugabyte-db/blob/master/architecture/design/docdb-automatic-tablet-splitting.md). While the broader feature is [work-in-progress](https://github.com/yugabyte/yugabyte-db/issues/1004), tablets can be automatically split for a few scenarions starting the v2.2 release. Detailed documentation is coming soon.
+Automatic tablet splitting enables resharding of data in a cluster automatically while online, and transparently to users, when a specified size threshold has been reached.
+
+For details on the architecture design, see [Automatic Re-sharding of Data with Tablet Splitting](https://github.com/yugabyte/yugabyte-db/blob/master/architecture/design/docdb-automatic-tablet-splitting.md). While the broader feature is [work-in-progress](https://github.com/yugabyte/yugabyte-db/issues/1004), tablets can be automatically split for a few scenarios starting in the v2.2 release.
```
Comment (bmatican) [inline docs/content/latest/architecture/docdb-sharding/tablet-splitting.md:226]: I would remove the last sentence. Better to just call out that the feature is in BETA, than say that `the broader feature is a work in progress`, since that sounds more like something that users would not be incentivized to even try...

File: docs/content/latest/architecture/docdb-sharding/tablet-splitting.md [hunk 11]
```diff
@@ -195,21 +206,94 @@ Tablet UUID                       Range                                 Leader
...
+The lower the value for the threshold size, the more tablets will exist with the same amount of data.
+
+### Example using a YCSB workload with automatic tablet splitting
+
+In the following example, a three-node cluster is created and uses a YCSB workload to demonstrate the use of automatic tablet splitting in a YSQL database. For details on using YCSB with YugabyteDB, see the [YCSB](../../../benchmark/ycsb-jdbc/) section in the Benchmark guide.
+
+1. Create a three-node cluster.
+
+```sh
+./bin/yb-ctl --rf=3 create --num_shards_per_tserver=1 --ysql_num_shards_per_tserver=1 --master_flags '"tablet_split_size_threshold=30000000"' --tserver_flags '"memstore_size_mb=10"'
```
Comment (bmatican) [inline docs/content/latest/architecture/docdb-sharding/tablet-splitting.md:241]: can put a small note here that we use `memstore_size_mb=10` to force flushing files more frequently and see the feature working more quickly

File: docs/content/latest/architecture/docdb-sharding/tablet-splitting.md [hunk 12]
```diff
@@ -195,21 +206,94 @@ Tablet UUID                       Range                                 Leader
...
+
+```sh
+diff -C1 after-load.json after-run.json | grep tablet_id | sort | uniq
+```
+
+8. The list of tablets accessed can be compared with `http://127.0.0.1:9000/tablets` to make sure no presplit tablets have been accessed.
+
+## Current tablet splitting limitations
+
+Presplitting tablets and automatic tablet splitting are in beta. To follow the work-in-progress on tablet splitting, see [GitHub #1004](https://github.com/yugabyte/yugabyte-db/issues/1004).
```
Comment (bmatican) [inline docs/content/latest/architecture/docdb-sharding/tablet-splitting.md:290]: this note here is much more ok, since it's clearly related to limitations, rather than upfront, before the user even gets to try it

File: docs/content/latest/architecture/docdb-sharding/tablet-splitting.md [hunk 13]
```diff
@@ -195,21 +206,94 @@ Tablet UUID                       Range                                 Leader
...
+
+Here are known limitations that are planned to be resolved in the next releases:
 
+* Presplit tablets remain in the system forever and are not deleted from the disk.
+* There is no upper bound on the number of tablets for the table when automatic tablet splitting is enabled.
+* During tablet splitting, client applications can get an error from the driver and need to retry the request.
+* If tablet splitting occurs during an ongoing distributed transaction, it could be aborted and need to be retried.
```
Comment (bmatican) [inline docs/content/latest/architecture/docdb-sharding/tablet-splitting.md:297]: `and would need to be retried`

File: docs/content/latest/architecture/docdb-sharding/tablet-splitting.md [hunk 14]
```diff
@@ -195,21 +206,94 @@ Tablet UUID                       Range                                 Leader
...
+* Because splitting of tablets that are not completely compacted is not yet implemented, tablets created by tablet splitting might be split after they reach the specified size threshold.
```
Comment (bmatican) [inline docs/content/latest/architecture/docdb-sharding/tablet-splitting.md:298]: since we already mention that these will be resolved eventually, we can reword this to something like

`Tablets created as a result of splitting cannot be split again, until they are fully compacted. This could mean that the split will happen much later than after the hit the user specified size threshold.`

File: docs/content/latest/architecture/docdb-sharding/tablet-splitting.md (hunk 2 above)
Comment (stevebang) [reply docs/content/latest/architecture/docdb-sharding/tablet-splitting.md]: added

File: docs/content/latest/architecture/docdb-sharding/tablet-splitting.md (hunk 7 above)
Comment (stevebang) [reply docs/content/latest/architecture/docdb-sharding/tablet-splitting.md:17]: right, that's what i meant when i said in slack that this page started off intending to discuss only runtime changes, but then added the presplitting and manual mechanisms. I'll spend a couple of minutes and make this clearer, but thought i'd get stuck here too long.

File: docs/content/latest/architecture/docdb-sharding/tablet-splitting.md (hunk 8 above)
Comment (stevebang) [reply docs/content/latest/architecture/docdb-sharding/tablet-splitting.md:53]: fixing -- got mixed up
